src/sage/combinat/specht_module.py

```diff
@@ -317,25 +615,385 @@ def _acted_upon_(self, x, self_on_left=False):
 
             EXAMPLES::
 
-                sage: SGA = SymmetricGroupAlgebra(QQ, 4)
-                sage: SM = SGA.specht_module([3,1])
+                sage: SGA = SymmetricGroupAlgebra(QQ, 5)
+                sage: SM = SGA.tabloid_module([2,2,1])
                 sage: SGA.an_element() * SM.an_element()
-                14*B[0] + 18*B[1] + 8*B[2]
+                2*T[{1, 5}, {2, 3}, {4}] + 2*T[{1, 5}, {2, 4}, {3}] + 3*T[{1, 5}, {3, 4}, {2}]
+                 + 12*T[{1, 2}, {3, 4}, {5}] + 15*T[{1, 2}, {3, 5}, {4}] + 15*T[{1, 2}, {4, 5}, {3}]
                 sage: 4 * SM.an_element()
-                8*B[0] + 8*B[1] + 12*B[2]
+                8*T[{1, 2}, {3, 4}, {5}] + 8*T[{1, 2}, {3, 5}, {4}] + 12*T[{1, 2}, {4, 5}, {3}]
             """
-            # Check for a scalar first
+            # first check for the base action
             ret = super()._acted_upon_(x, self_on_left)
             if ret is not None:
                 return ret
-            # Check if it is in the symmetric group algebra
+
+            if self_on_left:
+                return None
             P = self.parent()
-            if x in P._ambient or x in P._ambient.group():
-                if self_on_left:  # it is only a left module
-                    return None
-                else:
-                    return P.retract(P._ambient(x) * self.lift())
-            return None
+            if x in P._SGA:
+                return P.sum(c * (perm * self) for perm, c in x.monomial_coefficients().items())
+            if x in P._SGA.indices():
+                return P.element_class(P, {T.symmetric_group_action(x): c for T, c in self._monomial_coefficients.items()})
+
+
+class SpechtModuleTableauxBasis(SpechtModule):
+    r"""
+    A Specht module of a partition in the classical standard
+    tableau basis.
+
+    This is constructed as a `S_n`-submodule of the :class:`TabloidModule`
+    (also referred to as the standard module).
+
+    .. SEEALSO::
+
+        - :class:`SpechtModule` for the generic diagram implementation
+          constructed as a left ideal of the group algebra
+        - :class:`~sage.combinat.symmetric_group_representations.SpechtRepresentation`
+          for an implementation of the representation by matrices.
+    """
+    def __init__(self, ambient):
+        r"""
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: SGA = SymmetricGroupAlgebra(QQ, 5)
+            sage: SM = SGA.specht_module([2,2,1])
+            sage: TestSuite(SM).run()
+        """
+        self._diagram = ambient._shape
+        SymmetricGroupRepresentation.__init__(self, ambient._SGA)
+
+        from sage.combinat.symmetric_group_algebra import e
+        ambient_basis = ambient.basis()
```

Comment:
The ambient space is the whole tabloid module M^\lambda, right?

Reply:
Yes, that's correct.